docs/migration_guides/opflow_migration.rst

```diff
@@ -0,0 +1,1748 @@
+.. _qiskit.quantum_info.BaseOperator: https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/quantum_info/operators/base_operator.py
+
+The :class:`qiskit.opflow.OperatorBase` abstract class can be replaced with |qiskit.quantum_info.BaseOperator|_ ,
+keeping in mind that |qiskit.quantum_info.BaseOperator|_ is more generic than its opflow counterpart.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Opflow
+     - Alternative
+   * - :class:`qiskit.opflow.OperatorBase`
+     - |qiskit.quantum_info.BaseOperator|_
+
+..  attention::
+
+    Despite the similar class names, :class:`qiskit.opflow.OperatorBase` and
+    |qiskit.quantum_info.BaseOperator|_ are not completely equivalent to each other, and the transition
+    should be handled with care. Namely:
+
+    1. :class:`qiskit.opflow.OperatorBase` implements a broader algebra mixin. Some operator overloads that were
+    commonly used :mod:`~qiskit.opflow` (for example ``~`` for ``.adjoint()``) are not defined for
+    |qiskit.quantum_info.BaseOperator|_. You might want to check the specific
+    :mod:`~qiskit.quantum_info` subclass instead.
+
+    2. :class:`qiskit.opflow.OperatorBase` also implements methods such as ``.to_matrix()`` or ``.to_spmatrix()``,
+    which are only found in some of the |qiskit.quantum_info.BaseOperator|_ subclasses.
+
+    See API reference for more information.
+
+
+Operator Globals
+================
+*Back to* `Contents`_
+
+Opflow provided shortcuts to define common single qubit states, operators, and non-parametrized gates in the
+|operator_globals|_ module.
+
+These were mainly used for didactic purposes or quick prototyping, and can easily be replaced by their corresponding
+:mod:`~qiskit.quantum_info` class: :class:`~qiskit.quantum_info.Pauli`, :class:`~qiskit.quantum_info.Clifford` or
+:class:`~qiskit.quantum_info.Statevector`.
+
+
+1-Qubit Paulis
+--------------
+*Back to* `Contents`_
+
+The 1-qubit paulis were commonly used for quick testing of algorithms, as they could be combined to create more complex operators
+(for example, ``0.39 * (I ^ Z) + 0.5 * (X ^ X)``).
+These operations implicitly created operators of type  :class:`~qiskit.opflow.primitive_ops.PauliSumOp`, and can be replaced by
+directly creating a corresponding :class:`~qiskit.quantum_info.SparsePauliOp`, as shown in the examples below.
+
+
+.. list-table::
+   :header-rows: 1
+
+   * - Opflow
+     - Alternative
+   * - :class:`~qiskit.opflow.X`, :class:`~qiskit.opflow.Y`, :class:`~qiskit.opflow.Z`, :class:`~qiskit.opflow.I`
+     - :class:`~qiskit.quantum_info.Pauli`
+
+       ..  tip::
+
+           For direct compatibility with classes in :mod:`~qiskit.algorithms`, wrap in :class:`~qiskit.quantum_info.SparsePauliOp`.
+
+
+.. _1_q_pauli:
+
+.. raw:: html
+
+    <details>
+    <summary><a><font size="+1">Example 1: Defining the XX operator</font></a></summary>
+    <br>
+
+**Opflow**
+
+.. code-block:: python
+
+    >>> from qiskit.opflow import X
+
+    >>> operator = X ^ X
```

Comment:
I noticed that [you added](https://github.com/Qiskit/qiskit-terra/pull/9549/commits/fc715320e5143017622353c827073d91fdf2a54d) `>>>` for all code examples with intention to test the code with doctest. We need to do a few things before we can do that. Firstly, you need to change the directive from `code-block` to `doctest` or `testcode`. I personally prefer `testcode` because the code doesn't need to be written as an interactive python session like doctest style with `>>>` which is closer to what most people will use qiskit (that is in Jupyter notebook / python script). 

Secondly, doctest doesn't not test the code during docs build using the default building command (`sphinx-build -b`). It needs to run with a different commend (`sphinx-build -b doctest`, which only does testing, not building) to do doctest which Terra doesn't have currently. This is implemented in the sphinx theme for building the docs guide (see: https://github.com/Qiskit/qiskit_sphinx_theme/blob/2b5ced2061f842d3ac3424080d014cf028eb747a/tox.ini#L21-L27 and https://github.com/Qiskit/qiskit_sphinx_theme/blob/2b5ced2061f842d3ac3424080d014cf028eb747a/.github/workflows/docs-guide-deploy.yml#L25-L28) We plan to implement that in the future but it's out of the scope of this PR and it should not be blocked by that. 

See https://qiskit.github.io/qiskit_sphinx_theme/sphinx_guide/how_to_add_code.html#run-the-tests for more details.

That said, I actually think we shouldn't test the code in migration guides, because these code will be deprecated and will raise errors when they are removed if we test the code. And since we are not testing the code, I prefer not having `>>>` each code line. Sorry it might be a bit trouble for you to revert it back.

Reply:
It is possible to add skip directives, so its testable while the code is there - of course it would mean editing it once things are removed. For me it would be preferable to have this tested by CI to ensure it works and continues to do so as described until the code is removed.

Reply:
I agree. Thinking a bit more about it, we will anyway need to edit the migration guides after the features are removed because cross reference like `:class:`, `:meth:` will break.